test/functional/lua/vim_spec.lua

```diff
@@ -606,4 +606,13 @@ describe('lua stdlib', function()
     ]]
     eq(0, funcs.luaeval "vim.wo.cole")
   end)
+
+  it('vim.cmd', function()
+    exec_lua [[
+    vim.cmd.autocmd("BufNew * ++once lua BUF = vim.fn.expand('<abuf>')")
+    vim.cmd.new()
```

Comment:
in chat/IRC, did we agree to this API:

```
vim.cmd("autocmd BufNew * ++once lua BUF = vim.fn.expand('<abuf>')")
vim.cmd('new')
```

Because as @bfredl mentioned, Ex commands in Vim are not parsed in any predictable way, including range and bang (!). So to add a functional interface (e.g. `vim.cmd.new()`) for a command is a case-by-case task.

Reply:
I'm not sure anymore, tbh. `vim.cmd("autocmd` and `vim.cmd.autocmd(` are actually *exactly* the same number of characters, so I'm ok with that change. It's funny how it tricks my mind to think it's shorter.

Reply:
let's start with the no-sugar approach, can revisit later.